internal/lightcone/hunt/swordplay/swordplay.go

```diff
@@ -0,0 +1,89 @@
+package swordplay
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Check  key.Modifier = "swordplay_check"
+	Buff   key.Modifier = "swordplay_buff"
+	Target key.Modifier = "swordplay_target"
+)
+
+// For each time the wearer hits the same target, DMG dealt increases by
+// 8/10/12/14/16%, stacking up to 5 time(s). This effect will be dispelled when
+// the wearer changes targets.
+func init() {
+	lightcone.Register(key.Swordplay, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        4,
+		Path:          model.Path_HUNT,
+		Promotions:    promotions,
+	})
+
+	modifier.Register(Check, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnBeforeHit: onBeforeHit,
+			OnAfterHit:  onAfterHit,
+		},
+	})
+
+	modifier.Register(Buff, modifier.Config{
+		StatusType:        model.StatusType_STATUS_BUFF,
+		Stacking:          modifier.ReplaceBySource,
+		MaxCount:          5,
+		CountAddWhenStack: 1,
+		Listeners: modifier.Listeners{
+			OnAdd: buffOnAdd,
+		},
+	})
+
+	modifier.Register(Target, modifier.Config{})
+}
+
+func Create(engine engine.Engine, owner key.TargetID, lc info.LightCone) {
+	engine.AddModifier(owner, info.Modifier{
+		Name:   Check,
+		Source: owner,
+		State:  0.06 + 0.02*float64(lc.Imposition),
+	})
+}
+
+func onBeforeHit(mod *modifier.ModifierInstance, e event.HitStartEvent) {
+	if !mod.Engine().HasModifier(e.Defender, Target) {
```

Comment:
There is an interesting bug here which is more to do with the current set of tools the `engine` exposes. In the current state, if multiple people have swordplay equipped the targeting information would be "shared" between them. This could create multiple valid targets at the same time. Instead, each character with swordplay should have its own unique target.

The datamine solves this by having their `HasModifier` check also include the `Source`, which is not a check we have in engine. We may want to open an issue to add support for more complex `HasModifier` searches for cases like this.

You could write such a check in this implementation. It would look something like:
```go
func hasModifierFromSource(engine engine.Engine, key key.Modifier, target, source key.TargetID) bool {
  for _, mod := range mod.Engine().GetModifiers(target, key) {
    if mod.Source == source {
      return true
    }
  }
  return false
}
```